### ISSUE TITLE
 quantization aware training Pytorch and TensorRT

### ISSUE THREAD

**Issue by TranThanh96** [pytorch/pytorch#54748]
## ❓  quantization aware training Pytorch and TensorRT

Can I training model pytorch with "quantization aware training", then export to TensorRT INT8? 

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a @vkuzo

**Comment by supriyar** [comment]
This is currently not supported out of the box. Can you describe your use case? Are you trying to use ONNX to convert a PT quantized model to TRT model?

**Comment by vkuzo** [comment]
for triage

**Comment by TranThanh96** [comment]
> This is currently not supported out of the box. Can you describe your use case? Are you trying to use ONNX to convert a PT quantized model to TRT model?

cause I convert pytorch -> onnx -> trt int8. accuracy drop so much, So I think "quantization aware training" is good feature

**Comment by jianyuh** [comment]
@TranThanh96 Have you tried to use TRT FP16? How is accuracy impact?

**Comment by TranThanh96** [comment]
when training, I used Apex, so TRT FP16 does not make accuracy decrease

**Comment by vkuzo** [comment]
@jerryzh168 , would you mind sharing your thoughts on how PyTorch aims to solve this in the future?

**Comment by jerryzh168** [comment]
As of now, we can use PyTorch QAT and then FX Graph Mode Quantization + fx2trt (https://github.com/pytorch/pytorch/blob/master/torch/fx/experimental/fx2trt/example/quantized_resnet_test.py#L22) to run on TensorRT.

In the future we'll gradually complete https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/backend_config_dict/tensorrt.py for tensorrt and fully follow the recommendations for observer/fake_quant placement provided in TensorRT docs (e.g. don't insert observers for the output of weighted operations).

**Comment by jerryzh168** [comment]
Should we close this for now?

**Comment by songkq** [comment]
> 

@jerryzh168 Hi, could you please share a latest tutorial about how to use PyTorch QAT model  for TensorRT-INT8 deployment? Thanks.
